diff --git a/core/shared/platform/include/platform_wasi_types.h b/core/shared/platform/include/platform_wasi_types.h
index 0000000..0000000 100644
--- a/core/shared/platform/include/platform_wasi_types.h
+++ b/core/shared/platform/include/platform_wasi_types.h
@@ -157,7 +157,7 @@
 #if defined(_MSC_VER)
 #define ALIGNED_(x) __declspec(align(x))
 #define WARN_UNUSED _Check_return_
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) || defined(__TINYC__)
 #define ALIGNED_(x) __attribute__ ((aligned(x)))
 #define WARN_UNUSED __attribute__((__warn_unused_result__))
 #endif
